### PR TITLE
Populate README

### LINKED ISSUE #17: Populate README
The README should contain an explanation of what the repo is for and how to go about adding new tests/clients.

Waiting for #1.

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
Closes #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+`sigstore-conformance` is a conformance testing suite for Sigstore clients.
+
+## Supported clients
+
+The following clients are supported by `sigstore-conformance`:
+
+- [sigstore-python](https://github.com/sigstore/sigstore-python)
+- [cosign](https://github.com/sigstore/cosign)
+
+## Design
```
Comment (tetsuo-cpp) [inline README.md:17]: Not sure about this section. It's important information that needs to be communicated somewhere but it reads a bit disjointed.

Comment (di) [reply README.md:17]: I think this is fine for now -- the audience here is pretty small, just other Sigstore client maintainers.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+<!--- @begin-badges@ --->
+![CI](https://github.com/sigstore/sigstore-conformance/workflows/CI/badge.svg)
+<!--- @end-badges@ --->
+
+`sigstore-conformance` is a conformance testing suite for Sigstore clients.
+
+## Supported clients
+
+The following clients are supported by `sigstore-conformance`:
```
Comment (di) [inline README.md:12]:
```suggestion
The following clients are currently being tested by `sigstore-conformance`:
```

**Comment by tetsuo-cpp** [comment]
This is going to look significantly different so I'll just close this and give it another go.

**Pull request closed without merging**